src/ros1.ne

```diff
@@ -33,15 +27,31 @@ main ->
 
 # Types
 
-boolType -> %boolType {% function(d) { return { type: d[0].value } } %}
-
-numericType -> %numericType {% function(d) { return { type: d[0].value } } %}
-
-stringType -> %stringType {% function(d) { return { type: d[0].value } } %}
-
-timeType -> %timeType {% function(d) { return { type: d[0].value } } %}
-
-customType -> %fieldOrCustomType {% function(d) { return { type: d[0].value } } %}
+boolType -> "bool" {% function(d) { return { type: d[0].value } } %}
+
+numericType ->
+   ("byte"
+  | "char"
+  | "float32"
+  | "float64"
+  | "int8"
+  | "uint8"
+  | "int16"
+  | "uint16"
+  | "int32"
+  | "uint32"
+  | "int64"
+  | "uint64") {% function(d) { return { type: d[0][0].value } } %}
+
+stringType -> "string" {% function(d) { return { type: d[0].value } } %}
+
+timeType -> ("time" | "duration") {% function(d) { return { type: d[0][0].value } } %}
+
+customType -> %fieldOrType {% function(d, _, reject) {
+  const type = d[0].value;
+  if (type.match(/^bool$|^byte$|^char$|^float32$|^float64$|^int8$|^uint8$|^int16$|^uint16$|^int32$|^uint32$|^int64$|^uint64$|^string$|^time$|^duration$/) != undefined) return reject;
```

Comment:
Use Array.includes instead? Why do we need to reject here anyway?

Reply:
If you don't reject primitive types here you have an ambiguous parse of `bool x`. It would be both a primitive and a complex type.

Reply:
👍 and nearley will throw an error?

Reply:
No, nearley will give you both parses. Which is bad news for us.